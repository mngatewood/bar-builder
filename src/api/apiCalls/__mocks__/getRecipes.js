import mockFetchResponse from '../../../mockData/mockData';

export const getMovies = jest.fn().mockImplementation(() => {
  return Promise.resolve({ mockFetchResponse });
});
