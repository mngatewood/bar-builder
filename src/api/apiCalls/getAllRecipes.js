const urlRoot = "https://www.thecocktaildb.com/api/json/v1/1/";

export const getAlcoholicRecipes = async () => {
  const urlQuery = "filter.php?a=Alcoholic";
  const url = `${urlRoot}${urlQuery}`;
  try {
    const response = await fetch(url);
    const recipes = await response.json();
    return recipes.drinks;
  } catch (error) {
    throw Error("Error retrieving recipes");
  }
};



// export const getMovies = async () => {
//   const urlRoot = "https://api.themoviedb.org/3/movie/now_playing?";
//   const urlKey = `api_key=${apiKey}`;
//   const urlQuery = "&language=en-US&page=1";
//   const url = `${urlRoot}${urlKey}${urlQuery}`;

//   try {
//     const response = await fetch(url);
//     const movies = await response.json();
//     return movies.results;
//   } catch (error) {
//     throw new Error("Error getting movies");
//   }
// };
