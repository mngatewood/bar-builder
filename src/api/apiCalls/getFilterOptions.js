export const getFilterOptions = async (filterType) => {
  const urlRoot = 'https://www.thecocktaildb.com/api/json/v1/1/';
  const urlQuery = 'list.php?';
  const url = `${urlRoot}${urlQuery}${filterType}=list`;
  try {
    const response = await fetch(url);
    const filterOptions = await response.json();
    return filterOptions.drinks;
  } catch (error) {
    throw Error("Error retrieving options: " + error.message);
  }
};

