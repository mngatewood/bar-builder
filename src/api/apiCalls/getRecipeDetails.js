export const getRecipeDetails = (recipeArray) => {
  const urlRoot = 'https://www.thecocktaildb.com/api/json/v1/1/';
  const urlQuery = 'lookup.php?i=';
  const promises = recipeArray.map(async recipe => {
    try {
      const response = await fetch(`${urlRoot}${urlQuery}${recipe.idDrink}`);
      const recipeData = await response.json();
      const recipeDetails = await recipeData.drinks[0];
      return recipeDetails;
    } catch (error) {
      throw Error("Error retrieving recipe details");
    }
  });
  return Promise.all(promises);
};
