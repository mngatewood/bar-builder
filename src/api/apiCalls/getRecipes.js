const urlRoot = 'https://www.thecocktaildb.com/api/json/v1/1/';

export const getRecipes = async (filterType, filterValue) => {
  const urlQuery = 'filter.php?';
  const url = `${urlRoot}${urlQuery}${filterType}=${filterValue}`;
  try {
    const response = await fetch(url);
    const recipeSummary = await response.json();
    console.log(recipeSummary)
    const recipeDetails = await getRecipeDetails(recipeSummary.drinks);
    return recipeDetails;
  } catch (error) {
    throw Error("Error retrieving recipes");
  }
};

export const getRecipeDetails = (recipeArray) => {
  const urlQuery = 'lookup.php?i=';
  const promises = recipeArray.map( async recipe => {
    try {
      const response = await fetch(`${urlRoot}${urlQuery}${recipe.idDrink}`);
      const recipeData = await response.json();
      const recipeDetails = await recipeData.drinks[0];
      return recipeDetails;
    } catch (error) {
      throw Error("Error retrieving recipe details");
    }
  });
  return Promise.all(promises);
};

