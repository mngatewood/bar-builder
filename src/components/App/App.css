.App {
  text-align: center;
}

main {
  display: flex;
  overflow: hidden;
}