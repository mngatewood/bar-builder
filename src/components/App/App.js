import React, { Component } from 'react';
import { Route, withRouter } from 'react-router-dom';
import PropTypes from 'prop-types';
import './App.css';
import { getAlcoholicRecipes } from '../../api/apiCalls/getAllRecipes';


export class App extends Component {
  constructor() {
    super();
    this.state = {
      recipes: []
    };
  }

  async componentDidMount() {
    const recipes = await getAlcoholicRecipes();
    this.setState({ recipes: recipes });
  }

  render() {
    return (
      <div className="App">
        <header className="App-header">
          <h1 className="App-title">Welcome to The Bar Builder</h1>
        </header>
        <p className="App-intro">
        </p>
      </div>
    );
  }
}

export default App;
