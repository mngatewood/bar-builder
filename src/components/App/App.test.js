import React from 'react';
import App, { mapDispatchToProps } from './App';
import { shallow, mount } from 'enzyme';
import { getRecipes } from '../../api/apiCalls/getRecipes';
jest.mock('../../api/apiCalls/getRecipes');

describe('App', () => {

  it('matches the snapshot', () => {
    const wrapper = shallow(<App />);
    expect(wrapper).toMatchSnapshot();
  });

  it.skip('calls getRecipes with the correct params on componentDidMount ', () => {
    const wrapper = shallow(<App />);
    wrapper.instance().componentDidMount();
    expect(getRecipes).toHaveBeenCalled();
  });

  it('calls dispatch function when using mapDispatchToProps', () => {
    const mockDispatch = jest.fn();
    const mapped = mapDispatchToProps(mockDispatch);
    mapped.addRecipes();
    mapped.addCategories();
    mapped.addIngredients();
    mapped.addAlcoholicOptions();
    expect(mockDispatch).toHaveBeenCalled();
  });

});
