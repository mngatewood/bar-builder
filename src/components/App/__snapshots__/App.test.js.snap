// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App matches the snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <App
    addAlcoholicOptions={[Function]}
    addCategories={[Function]}
    addIngredients={[Function]}
    addRecipes={[Function]}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <header>
          <withRouter(Connect(Header)) />
</header>,
        <main>
          <Route
                    component={[Function]}
                    exact={true}
                    path="/"
          />
          <Route
                    component={[Function]}
                    exact={true}
                    path="/recipes"
          />
          <Route
                    component={[Function]}
                    exact={true}
                    path="/inventory"
          />
          <Route
                    component={[Function]}
                    exact={true}
                    path="/menu"
          />
</main>,
        <footer>
          <withRouter(Connect(Dashboard)) />
</footer>,
      ],
      "className": "App",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <withRouter(Connect(Header)) />,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        "type": "header",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <Route
              component={[Function]}
              exact={true}
              path="/"
/>,
            <Route
              component={[Function]}
              exact={true}
              path="/recipes"
/>,
            <Route
              component={[Function]}
              exact={true}
              path="/inventory"
/>,
            <Route
              component={[Function]}
              exact={true}
              path="/menu"
/>,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "component": [Function],
              "exact": true,
              "path": "/",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "component": [Function],
              "exact": true,
              "path": "/recipes",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "component": [Function],
              "exact": true,
              "path": "/inventory",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "component": [Function],
              "exact": true,
              "path": "/menu",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": "main",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <withRouter(Connect(Dashboard)) />,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        "type": "footer",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <header>
            <withRouter(Connect(Header)) />
</header>,
          <main>
            <Route
                        component={[Function]}
                        exact={true}
                        path="/"
            />
            <Route
                        component={[Function]}
                        exact={true}
                        path="/recipes"
            />
            <Route
                        component={[Function]}
                        exact={true}
                        path="/inventory"
            />
            <Route
                        component={[Function]}
                        exact={true}
                        path="/menu"
            />
</main>,
          <footer>
            <withRouter(Connect(Dashboard)) />
</footer>,
        ],
        "className": "App",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <withRouter(Connect(Header)) />,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {},
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": "header",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <Route
                component={[Function]}
                exact={true}
                path="/"
/>,
              <Route
                component={[Function]}
                exact={true}
                path="/recipes"
/>,
              <Route
                component={[Function]}
                exact={true}
                path="/inventory"
/>,
              <Route
                component={[Function]}
                exact={true}
                path="/menu"
/>,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "component": [Function],
                "exact": true,
                "path": "/",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "component": [Function],
                "exact": true,
                "path": "/recipes",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "component": [Function],
                "exact": true,
                "path": "/inventory",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "component": [Function],
                "exact": true,
                "path": "/menu",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
          ],
          "type": "main",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <withRouter(Connect(Dashboard)) />,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {},
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": "footer",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
